upgrade/steps.go

```diff
@@ -140,7 +140,17 @@ func ensureUpstreamRepo(ctx Context, repoPath string) step.Step {
 
 			// from github.com/org/repo to $GOPATH/src/github.com/org
 			expectedLocation = filepath.Join(strings.Split(repoPath, "/")...)
-			expectedLocation = filepath.Join(ctx.GoPath, "src", expectedLocation)
+			cwd, err := os.Getwd()
+			if err != nil {
+				return "", fmt.Errorf("could not resolve current directory: '%s", err)
+			}
+			if idx := strings.Index(cwd, filepath.Base(repoPath)); idx != -1 {
+				idx = idx + len(expectedLocation) - 1
+				expectedLocation = cwd[:idx]
+			} else {
+				expectedLocation = filepath.Join(ctx.GoPath, "src", expectedLocation)
+
+			}
```

Comment:
Can you explain what this is trying to do. It seems like the logic should be
```go
if filepath.Base(repoPath) == filepath.Base(cwd) && filepath.Base(cwd) != "" {
    	expectedLocation = cwd
else {
    	expectedLocation = filepath.Join(ctx.GoPath, "src", expectedLocation)
}
```

Reply:
I think I was trying to account for the possibility of being in a subdirectory, i.e. `user/home/pulumi/pulumi-docker/provider`would set the expected location to `user/home/pulumi/pulumi-docker`